ttt-online/src/models/game.py

```diff
@@ -83,12 +93,29 @@ def update(self, coords: t.List[int]):
         self.field[row][col] = self._current_mark()
         self.step_count += 1
         if self.step_count >= math.ceil(field_size / 2) and self._is_win():
-            self.winner_id = self.current_player_id
+            self.winner = self.current_player
             self.current_state = State.DONE.value
             return
         self._invert_player()
         if self.step_count == field_size:
             self.current_state = State.DONE.value
 
-    def to_json(self):
-        return self.__dict__
+    @classmethod
+    def from_dict(cls, data: dict):
+        players_mark = ['X' if data[f'{prefix}_id'] == data['owner_id'] else '0' for prefix in cls.player_prefixes]
+
+        for idx, prefix in enumerate(cls.player_prefixes):
+            id = data.pop(f'{prefix}_id')
+            name = data.pop(f'{prefix}_name')
+            mark = players_mark[idx]
+            if id is None:
+                mark = None
+            data[prefix] = Player(id=id, name=name, mark=mark)
```

Comment:
If prefix_id is none - then data[prefix] must be None, am i right?
And then, you can not to do checks in `to_dict` method